features/step_definitions/kata.rb

```diff
@@ -1,229 +1,103 @@
-    end
-  end
-end
-
-Given /^I wait for #{QUOTED} (uninstall|install) to start$/ do | kc_name, mode |
-  timeout = 300
-  stats = {}
-  success = false
-  wait_for(timeout, stats: stats) do
-    if mode == 'install'
-      success = kata_config(kc_name).installing?(user: user, quiet: true, cached: false)
-    else
-      success = kata_config(kc_name).uninstalling?(user: user, quiet: true, cached: false)
-    end
-    break if success
-  end
-  unless success
-    raise "#{mode} failed to start after #{timeout} seconds"
-  else
-    logger.info("#{mode} started...")
-  end
-end
-
-Given /^I wait until number of completed kata runtime nodes match#{OPT_QUOTED} for #{QUOTED}$/ do |number, kc_name|
+Given /^I wait until number of completed kata runtime nodes for #{QUOTED} matches$/ do |kc_name|
   ready_timeout = 1200
   matched = kata_config(kc_name).wait_till_installed_counter_match(
     user: user, seconds: ready_timeout)
-  unless matched[:success]
-    installed_node_count = kata_config(kc_name).install_completed_node_count(user: user, cached: false)
-    expected_node_count = kata_config(kc_name).total_nodes_count
-    raise "Kata runtime did not install into all worker nodes, only #{installed_node_count} reached, expecting #{expected_node_count}"
-  end
+  installed_node_count = kata_config(kc_name).install_completed_node_count(user: user, cached: false)
+  expected_node_count = kata_config(kc_name).total_nodes_count
+  raise "Kata runtime did not install into all worker nodes, only #{installed_node_count} reached, expecting #{expected_node_count}" unless matched[:success]
+  logger.info("#{installed_node_count} out of #{expected_node_count} worker nodes installed with kata runtime")
 end
 
-Given /^I remove kata operator from the#{OPT_QUOTED} namespace$/ do | kata_ns |
-  ensure_admin_tagged
-  step %Q/I store master major version in the clipboard/
-  kata_ns ||= "openshift-sandboxed-containers-operator"
-  step %Q/I switch to cluster admin pseudo user/
-  # 1. remove kataconfig first
-  project(kata_ns)
-  # prereq is that there are no outstanding pods in the cluster with
-  # kata-runtime.  Do the destructive action of removing them so kataconfig can be removed.
-  step %Q/I find all pods running with kata as runtime in the cluster and store them to the :kata_pods clipboard/
-  if cb.kata_pods.count > 0
-    logger.info("All existing pods using kata must be removed before kataconfig can be uninstalled.")
-    logger.info("Removing all pods using kata...")
-    step %Q/I remove all kata pods in the cluster stored in the clipboard/
-  end
-
-  kataconfig_name = BushSlicer::KataConfig.list(user: admin).first.name
-  step %Q/I ensure "#{kataconfig_name}" kata_config is deleted within 1500 seconds/
-  # 2. remove namespace
-  step %Q/I ensure "#{kata_ns}" project is deleted/
+Given /^Valid cluster type for kata tests exists$/ do
+  accepted_platforms = ['gcp', 'azure']
```

Comment:
@valtair1891 doesn't kata support baremetal install as well?

Reply:
It does, it will be added, after testing and debugging, since it's not stable